### PR TITLE
MpidManager sending PutFailure: outbox or inbox full

### PR DESCRIPTION
**Pull request by maqi**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/safe_vault/323)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @Fraser999

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/personas/mpid_manager.rs [hunk 1]
```diff
@@ -198,6 +198,9 @@ impl MpidManager {
                        .or_insert(Account::default())
                        .put_into_inbox(data.payload_size() as u64, &data.name(), &None) {
                     let _ = self.chunk_store_inbox.put(&data.name(), data.value());
+                } else {
+                    let _ = routing_node.send_put_failure(request.dst.clone(),
+                        request.src.clone(), request.clone(), Vec::new(), message_id);
```
Comment (brian-js) [inline src/personas/mpid_manager.rs]: Prefer using something like `try` to `let _ =`

File: src/personas/mpid_manager.rs [hunk 2]
```diff
@@ -237,6 +240,44 @@ impl MpidManager {
...
+                }
+            }
+            _ => unreachable!("Error in vault demuxing"),
+        }
+        Ok(())
+    }
+
+    // PutFailure only happens from receiver's MpidManager to sender's MpidManager to
+    // indicate an inbox full.
```
Comment (brian-js) [inline src/personas/mpid_manager.rs]: PutFailure is returned from MpidManager to Client, if Clients' Outbox or Inbox is full.

Comment (maqi) [inline src/personas/mpid_manager.rs]: This comment is about a PutFailure between mpid_managers.
the one you mentioned is to be handled by client

**Pull request merged**